onnxruntime/core/graph/contrib_ops/bert_defs.cc

```diff
@@ -1128,6 +1128,16 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
                "2D tensor with shape (max_sequence_length, head_size / 2).",
                "T",
                OpSchema::Optional)
+        .Input(9,
+               "custom_pos_ids",
+               "2D tensor with shape (batch_size, sequence_length).",
+               "tensor(int64)",
```

Comment:
Let's keep this as int32 to be consistent with the other input types to this op.

https://github.com/microsoft/onnxruntime/blob/9dec0564104cec8c9b13a631de0c3ad3c7feb0fb/onnxruntime/core/graph/contrib_ops/bert_defs.cc#L1157-L1158

```suggestion
               "M",
```

Reply:
Rotary embeddings kernel expects position ids to be of type int64_t so it has to be of this type (and also if we don't pass in any position ids, the default generated position ids withing the GQA op will be of type int64_t). If we were to change this to int32_t we would have an additional step to cast each position id to int64_t.